manifests/init.pp

```diff
@@ -467,6 +475,8 @@
   create_resources('elasticsearch::script', $elasticsearch::scripts)
   create_resources('elasticsearch::snapshot_repository', $elasticsearch::snapshot_repositories)
   create_resources('elasticsearch::template', $elasticsearch::templates)
+  create_resources('elasticsearch::component_template', $elasticsearch::component_templates)
```

Comment:
instead of `create_resources()`, could you iterate on the hash like:
```puppet
$elasticsearch::component_templates.each |String $key, Hash $values| {
  elasticsearch::component_template { $key:
    * => $values,
}
```

that pattern is now preferred over `create_resources()`